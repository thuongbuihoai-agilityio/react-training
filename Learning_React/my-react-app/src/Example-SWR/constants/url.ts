const BASE_URL = "http://localhost:1234";
const TASK_URL = `${BASE_URL}/task`;

export { BASE_URL, TASK_URL };
