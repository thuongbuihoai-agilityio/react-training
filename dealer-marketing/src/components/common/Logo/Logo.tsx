import React from "react";
import Image from "next/image";

const Logo: React.FC = () => {
  return (
    <figure>
      <a href="/">
        <Image
          src="/images/logos/logo-dealer-marketing.svg"
          alt="This is logo"
          width={256}
          height={43}
        />
      </a>
    </figure>
  );
};
export default Logo;
