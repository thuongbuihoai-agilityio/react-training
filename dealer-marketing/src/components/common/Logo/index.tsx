import Image from "next/image";
import Link from "next/link";
import React, { memo } from "react";

interface LogoProps {
  url: string;
}

const Logo: React.FC<LogoProps> = ({ url }) => (
  <div data-testid="logo">
    <Link data-testid="logo" href="/" passHref>
      <Image src={url} alt="This is logo" width={256} height={43} />
    </Link>
  </div>
);

export default memo(Logo);
