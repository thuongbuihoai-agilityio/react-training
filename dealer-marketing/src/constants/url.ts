const BASE_URL = process.env.NEXT_PUBLIC_ENDPOINTS;
const BLOGS_URL = `${BASE_URL}/api/blogs`;
const EXPERTS_URL = `${BASE_URL}/api/experts`;

const API_BLOGS = "/api/blogs";

export { BLOGS_URL, EXPERTS_URL, API_BLOGS };
