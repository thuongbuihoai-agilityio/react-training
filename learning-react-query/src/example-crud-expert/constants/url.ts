const EXPERT_URL = "https://63183dc9f6b281877c66cbe0.mockapi.io/api/experts";

export {
  EXPERT_URL,
};
