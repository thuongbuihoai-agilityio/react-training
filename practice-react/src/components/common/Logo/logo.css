@media screen and (min-width: 1366px) {
  .logo {
    width: 189px;
    height: 74px;
  }
}