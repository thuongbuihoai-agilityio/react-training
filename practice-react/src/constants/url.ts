const BASE_URL = "http://localhost:1234";
const PRODUCTS_URL = `${BASE_URL}/products`;
const CATEGORIES_URL = `${BASE_URL}/categories`;

export { BASE_URL, PRODUCTS_URL, CATEGORIES_URL };