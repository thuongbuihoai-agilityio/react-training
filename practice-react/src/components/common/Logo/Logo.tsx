import React from "react";
import { Link } from "react-router-dom";
import "./logo.css";

interface LogoProps {
  src: string;
  type?: string;
}

const Logo: React.FC<LogoProps> = ({ src, type }) => {
  let className = "";
  switch (type) {
    case "medium":
      className = "logo--medium";
      break;
    case "large":
      className = "logo--large";
      break;
    default:
      break;
  }

  return (
    <Link data-testid="logo" to="/">
      <img className={className} src={src} alt="This is logo" />
    </Link>
  );
};

export default Logo;
