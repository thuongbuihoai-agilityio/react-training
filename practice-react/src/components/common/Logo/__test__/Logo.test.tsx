import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import url from "../../../../assets/images/logos/logo.png";
import Logo from "../Logo";

describe("Logo component", () => {
  test("should render logo component", () => {
    const { getByTestId } = render(<Logo src={url} />);
    expect(getByTestId("logo")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Logo src={url} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
