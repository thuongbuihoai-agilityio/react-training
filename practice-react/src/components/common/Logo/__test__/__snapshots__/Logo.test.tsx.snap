// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component matches snapshot 1`] = `
<DocumentFragment>
  <img
    alt="This is logo"
    class="logo"
    data-testid="logo"
  />
</DocumentFragment>
`;
